python-packages/order_utils/src/zero_ex/order_utils/__init__.py

```diff
@@ -159,7 +182,209 @@ def int_to_32_big_endian_bytes(i: int):
+    """Parse signature hex as a concatentation of EC parameters ordered V, R, S.
+
+    >>> _parse_signature_hex_as_vrs('0x1b117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d872871137feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b03')
+    {'v': 27, 'r': '117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d87287113', 's': '7feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b'}
+    """  # noqa: E501 (line too long)
+    signature: ECSignature = {
+        "v": int(signature_hex[2:4], 16),
+        "r": signature_hex[4:68],
+        "s": signature_hex[68:132],
+    }
+    if signature["v"] == 0 or signature["v"] == 1:
+        signature["v"] = signature["v"] + 27
+    return signature
+
+
+def _parse_signature_hex_as_rsv(signature_hex: str) -> ECSignature:
+    """Parse signature hex as a concatentation of EC parameters ordered R, S, V.
+
+    >>> _parse_signature_hex_as_rsv('0x117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d872871137feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b00')
+    {'r': '117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d87287113', 's': '7feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b', 'v': 27}
+    """  # noqa: E501 (line too long)
+    signature: ECSignature = {
+        "r": signature_hex[2:66],
+        "s": signature_hex[66:130],
+        "v": int(signature_hex[130:132], 16),
+    }
+    if signature["v"] == 0 or signature["v"] == 1:
+        signature["v"] = signature["v"] + 27
+    return signature
+
+
+def _convert_ec_signature_to_vrs_hex(signature: ECSignature) -> str:
+    """Convert elliptic curve signature object to hex hash string.
+
+    >>> _convert_ec_signature_to_vrs_hex(
+    ...     {
+    ...         'r': '117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d87287113',
+    ...         's': '7feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b',
+    ...         'v': 27
+    ...     }
+    ... )
+    '0x1b117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d872871137feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b'
+    """  # noqa: E501 (line too long)
+    return (
+        "0x"
+        + signature["v"].to_bytes(1, byteorder="big").hex()
+        + signature["r"]
+        + signature["s"]
+    )
+
+
+def sign_order_hash(
+    provider: Web3.HTTPProvider, signer_address: str, order_hash_hex: str
+) -> str:
+    """Sign a message with the order hash, and return the signature.
+
+    :param provider: A Web3 provider.
+    :param signer_address: The address of the signing account.
+    :param order_hash_hex: A hex string representing the order hash, like that
+        returned from `generate_order_hash_hex()`.
+    :rtype: A string, of ASCII hex digits, representing the signature.
+
+    >>> sign_order_hash(
+    ...     Web3.HTTPProvider("http://127.0.0.1:8545"),
+    ...     Web3(Web3.HTTPProvider("http://127.0.0.1:8545")).personal.listAccounts[0],
```

Comment:
Thanks for noticing this.  It was indeed weird.  I had tried to do this once and it didn't work but I'm glad I tried again because now it's working great and I'm happy with it.